File: CS/WindowsApplication1/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;
using DevExpress.XtraTreeList.Nodes;
using DevExpress.XtraTreeList;

namespace WindowsApplication1
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }

        private void Form1_Load(object sender, EventArgs e)
        {
            dataSet1.DataTable1.AddDataTable1Row("row1", "row2");
        }

        private void button1_Click(object sender, EventArgs e)
        {
            string[] rows = GetClipboardRows();
            if (rows != null)
                AddRowsToTreeList(rows);
        }

        private void AddRowsToTreeList(string[] rows)
        {
            for (int i = 0; i < rows.Length; i++)
            {
                TreeListNode node = AddNewTreeListNode(treeList1);
                node.SetValue("Column1",rows[i]);
            }
            treeList1.ExpandAll();
        }

        private string[] GetClipboardRows()
        {
            if (Clipboard.GetDataObject().GetDataPresent(typeof(string)))
            {

                string text = Clipboard.GetText();

                if (!String.IsNullOrEmpty(text))
                {

                    string[] s = text.Split(Environment.NewLine.ToCharArray(),StringSplitOptions.RemoveEmptyEntries);
                    return s;
                }

            }
            return null;
        }


        public TreeListNode AddNewTreeListNode(TreeList treeList)
        {
            //RaiseInputNullExceptionIfNull(treeList, "treeList");

            TreeListNode parentTreeListNode = treeList.FocusedNode ?? treeList.Nodes.FirstNode;

            object[] obj = new object[treeList.Columns.Count];
            TreeListNode newTreeListNode = treeList.AppendNode(obj, parentTreeListNode);
            return newTreeListNode;
            //treeList.FocusedNode = newTreeListNode;
            //treeList.ExpandAll();
        }

    }
}

File: VB/WindowsApplication1/Form1.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.Text
Imports System.Windows.Forms
Imports DevExpress.XtraTreeList.Nodes
Imports DevExpress.XtraTreeList

Namespace WindowsApplication1
	Partial Public Class Form1
		Inherits Form
		Public Sub New()
			InitializeComponent()
		End Sub

		Private Sub Form1_Load(ByVal sender As Object, ByVal e As EventArgs) Handles MyBase.Load
			dataSet1.DataTable1.AddDataTable1Row("row1", "row2")
		End Sub

		Private Sub button1_Click(ByVal sender As Object, ByVal e As EventArgs) Handles button1.Click
			Dim rows() As String = GetClipboardRows()
			If rows IsNot Nothing Then
				AddRowsToTreeList(rows)
			End If
		End Sub

		Private Sub AddRowsToTreeList(ByVal rows() As String)
			For i As Integer = 0 To rows.Length - 1
				Dim node As TreeListNode = AddNewTreeListNode(treeList1)
				node.SetValue("Column1",rows(i))
			Next i
			treeList1.ExpandAll()
		End Sub

		Private Function GetClipboardRows() As String()
			If Clipboard.GetDataObject().GetDataPresent(GetType(String)) Then

				Dim text As String = Clipboard.GetText()

				If (Not String.IsNullOrEmpty(text)) Then

					Dim s() As String = text.Split(Environment.NewLine.ToCharArray(),StringSplitOptions.RemoveEmptyEntries)
					Return s
				End If

			End If
			Return Nothing
		End Function


		Public Function AddNewTreeListNode(ByVal treeList As TreeList) As TreeListNode
			'RaiseInputNullExceptionIfNull(treeList, "treeList");

			Dim parentTreeListNode As TreeListNode
            If Not (treeList.FocusedNode Is Nothing) Then
                parentTreeListNode = treeList.FocusedNode
            Else
                parentTreeListNode = treeList.Nodes.FirstNode
            End If

			Dim obj(treeList.Columns.Count - 1) As Object
			Dim newTreeListNode As TreeListNode = treeList.AppendNode(obj, parentTreeListNode)
			Return newTreeListNode
			'treeList.FocusedNode = newTreeListNode;
			'treeList.ExpandAll();
		End Function

	End Class
End Namespace